src/Bicep.LangServer.IntegrationTests/SemanticTokenTests.cs

```diff
@@ -75,6 +76,32 @@ public async Task Overlapping_tokens_are_not_returned(DataSet dataSet)
             }
         }
 
+        [DataTestMethod]
+        [DynamicData(nameof(GetParamsData), DynamicDataSourceType.Method)]
+        public async Task Correct_semantic_tokens_are_returned_for_params_file(string text, TextSpan[] spans)
+        {
+            var uri = new Uri($"file://{TestContext.TestName}_{Guid.NewGuid():D}/main.bicepparam");
+            var helper = await DefaultServer.GetAsync();
+            await helper.OpenFileOnceAsync(TestContext, text, uri);
+
+            var semanticTokens = await helper.Client.TextDocument.RequestSemanticTokens(new SemanticTokensParams
+            {
+                TextDocument = new TextDocumentIdentifier(uri),
+            });
+
+            var lineStarts = TextCoordinateConverter.GetLineStarts(text);
+            var tokenSpans = CalculateTokenTextSpans(lineStarts, semanticTokens!.Data).ToArray();
+
+            for (var i = 0; i < tokenSpans.Length; i++)
```

Comment:
Only able to verify the text spans of the semantic token. Is there a way to get the actual token type from the raw data?

Reply:
See docs under https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens, and look at the implementation of `CalculateTokenTextSpans`. You should be able to get `tokenType` & `tokenModifiers` by looking at the 4th (`i % 5 == 3`) & 5th (`i % 5 == 4`) integer in the data array.

Reply:
You could maybe use this as the lookup: https://github.com/OmniSharp/csharp-language-server-protocol/blob/7e0eefd220435ace5b8f6f3d0534a644394072f9/src/Protocol/Features/Document/SemanticTokensFeature.cs#L827-L831